### PR TITLE
pkp/pkp-lib#3101 Fix language settings on language removal

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php [hunk 1]
```diff
@@ -416,47 +417,33 @@ function setPrimaryLocale($args, $request) {
...
-		$siteDao = DAORegistry::getDAO('SiteDAO');
-		$siteDao->updateObject($site);
-
-		$this->_updateContextLocaleSettings($request);
-	}
-
 	/**
 	 * Helper function to update locale settings in all
-	 * installed contexts, based on site locale settings.
+	 * installed journals, based on site locale settings.
```
Comment (bozana) [inline controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php]: hmmm.. I believe the context should stay here? -- it is in the pkp-lib

File: controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php [hunk 2]
```diff
@@ -416,47 +417,33 @@ function setPrimaryLocale($args, $request) {
 	//
 	// Helper methods.
 	//
-	/**
-	 * Update the locale support state (enabled or disabled).
-	 * @param $request Request
-	 * @param $rowId string The locale row id.
-	 * @param $enable boolean Enable locale flag.
-	 */
-	function _updateLocaleSupportState($request, $rowId, $enable) {
```
Comment (bozana) [inline controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php]: Hmmm... this function is used in enableLocale and disableLocale function...

File: controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php [hunk 3]
```diff
@@ -416,47 +417,33 @@ function setPrimaryLocale($args, $request) {
...
+		$contexts = $contextDao->getAll();
+		while ($context = $contexts->next()) {
+			$primaryLocale = $context->getPrimaryLocale();
+			foreach (array('supportedLocales', 'supportedFormLocales', 'supportedSubmissionLocales') as $settingName) {
+				$localeList = $context->getSetting($settingName);
+
+				if (isset($primaryLocale) && !in_array($primaryLocale, $siteSupportedLocales)) {
+					$context->setPrimaryLocale($site->getPrimaryLocale());
+					$this->updateContext($context);
+				}
```
Comment (bozana) [inline controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php]: This if statement does not have to be in the foreach loop.
Also $this->updateContext does not exist, I think it should be $contextDao?

File: controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php (hunk 2 above)
Comment (asmecher) [reply controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php]: Oops -- adding again, thanks.

File: controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php (hunk 1 above)
Comment (asmecher) [reply controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php]: Yes, correcting. (This method was copied over from the OJS repo.)

File: controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php (hunk 3 above)
Comment (asmecher) [reply controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php]: I think that was existing code moved from elsewhere, but yes, I'll fix it.

**Comment by asmecher** [comment]
Thanks, @bozana, I've corrected the issues you found. Could you take another look?

**Comment by bozana** [comment]
@asmecher, yes, fine, you can merge... :-)

**Pull request merged**